### PR TITLE
Show PDF, PNG and JPEG files in projector

### PR DESCRIPTION
**Pull request by boehlke**
This PR is work in progress, clarification about projector controls required.

### CODE REVIEW

File: client/src/app/slides/mediafiles/mediafile/mediafile-slide.component.ts [hunk 1]
```diff
@@ -12,4 +14,8 @@ export class MediafileSlideComponent extends BaseSlideComponent<MediafileSlideDa
     public constructor() {
         super();
     }
+
+    @computed get url() {
```
Comment (FinnStutzenstein) [inline client/src/app/slides/mediafiles/mediafile/mediafile-slide.component.ts]: @boehlke For what case is the `@computed` there?

Comment (boehlke) [reply client/src/app/slides/mediafiles/mediafile/mediafile-slide.component.ts]: It is used in the componten.html...

Comment (FinnStutzenstein) [reply client/src/app/slides/mediafiles/mediafile/mediafile-slide.component.ts]: Is the computed necessary there?

Comment (boehlke) [reply client/src/app/slides/mediafiles/mediafile/mediafile-slide.component.ts]: Hi Finn, I am not sure... please decide for youself... I am not very friends with angular.
I can delete it if you want.

**Comment by boehlke** [comment]
Hi @FinnStutzenstein 
kannst Du vielleicht diese Probleme lösen?
https://travis-ci.org/OpenSlides/OpenSlides/jobs/497504963#L1222
Ich muss erstmal mit Angular warm werde und mir würde helfen mal zu sehen, wie man Angular/TS da glücklich macht.
Danke!

File: client/src/app/slides/mediafiles/mediafile/mediafile-slide.component.ts [hunk 2]
```diff
@@ -12,4 +13,8 @@ export class MediafileSlideComponent extends BaseSlideComponent<MediafileSlideDa
     public constructor() {
         super();
     }
+
+    @computed public get url(): string {
```
Comment (FinnStutzenstein) [inline client/src/app/slides/mediafiles/mediafile/mediafile-slide.component.ts]: TODO: remove

**Comment by FinnStutzenstein** [comment]
@boehlke Die modulkonfigurationen müssen statisch sein, denn diese werden durch Angular beim Start analysiert -> kein `.push` an dieser Stelle.
Ich hab über deine kleinen Änderungne drübergepushed. Dort sind jetzt erste controls in der UI (prpojector-detail). Letzte/Nächste seite für PDFs funktioniert und wird in das projector element geschrieben. Zugriff auf diese Daten erhälst du in der Slide mit `this.data.element(.page, .rotation, ...)`. Jedoch immer darauf achten, dass diese nicht zwangsweise gesetzt sein müssen und dann von defaults ausgehen (z.B. page=1).

Die Steuerung muss noch mit zoom/rotation erweitert werden. Da kannst du dich an `.pdfSetPage` orientieren.

**Comment by FinnStutzenstein** [comment]
Ich bekomme manchmal folgenden Fehler:
```
core.js:15714 ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'div' of undefined
TypeError: Cannot read property 'div' of undefined
    at backtrackBeforeAllVisibleElements (pdf_viewer.js:637)
    at getVisibleElements (pdf_viewer.js:684)
    at PDFViewer._getVisiblePages (pdf_viewer.js:6435)
    at PDFViewer.forceRendering (pdf_viewer.js:6065)
    at PDFRenderingQueue.renderHighestPriority (pdf_viewer.js:5162)
    at continueRendering (pdf_viewer.js:5227)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:391)
    at Object.onInvoke (core.js:17289)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:390)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.run (zone.js:150)
    at backtrackBeforeAllVisibleElements (pdf_viewer.js:637)
    at getVisibleElements (pdf_viewer.js:684)
    at PDFViewer._getVisiblePages (pdf_viewer.js:6435)
    at PDFViewer.forceRendering (pdf_viewer.js:6065)
    at PDFRenderingQueue.renderHighestPriority (pdf_viewer.js:5162)
    at continueRendering (pdf_viewer.js:5227)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:391)
    at Object.onInvoke (core.js:17289)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:390)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.run (zone.js:150)
    at resolvePromise (zone.js:831)
    at zone.js:896
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:423)
    at Object.onInvokeTask (core.js:17280)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:422)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.runTask (zone.js:195)
    at drainMicroTaskQueue (zone.js:601)
    at ZoneTask.push../node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask [as invoke] (zone.js:502)
    at invokeTask (zone.js:1744)
    at WebSocket.globalZoneAwareCallback (zone.js:1770)
```

File: client/src/app/slides/mediafiles/mediafile/mediafile-slide.component.ts (hunk 2 above)
Comment (FinnStutzenstein) [reply client/src/app/slides/mediafiles/mediafile/mediafile-slide.component.ts]: (`@computed`)

**Comment by FinnStutzenstein** [comment]
Todo (for me): Document the new functions.

**Comment by FinnStutzenstein** [comment]
@boehlke Add `PdfViewerModule` to the `mediafile-slide.component.spec.ts` `imports`-list.

**Comment by FinnStutzenstein** [comment]
@boehlke kannst gerne weitermachen

**Comment by emanuelschuetze** [comment]
Feedback of my first test:
- PDF (and images) should project always in fullscreen (use best height/width). Rotation/Zoom/fullscreen controls are not important for me.
- hide page controls if pdf has one page only
- wrong z-index (header/footer in front)
- please rebase on current master to test with updated projector-detail template

**Comment by boehlke** [comment]
> Ich bekomme manchmal folgenden Fehler:
> 
> ```
> core.js:15714 ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'div' of undefined
> TypeError: Cannot read property 'div' of undefined
>     at backtrackBeforeAllVisibleElements (pdf_viewer.js:637)
>     at getVisibleElements (pdf_viewer.js:684)
>     at PDFViewer._getVisiblePages (pdf_viewer.js:6435)
>     at PDFViewer.forceRendering (pdf_viewer.js:6065)
>     at PDFRenderingQueue.renderHighestPriority (pdf_viewer.js:5162)
>     at continueRendering (pdf_viewer.js:5227)
>     at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:391)
>     at Object.onInvoke (core.js:17289)
>     at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:390)
>     at Zone.push../node_modules/zone.js/dist/zone.js.Zone.run (zone.js:150)
>     at backtrackBeforeAllVisibleElements (pdf_viewer.js:637)
>     at getVisibleElements (pdf_viewer.js:684)
>     at PDFViewer._getVisiblePages (pdf_viewer.js:6435)
>     at PDFViewer.forceRendering (pdf_viewer.js:6065)
>     at PDFRenderingQueue.renderHighestPriority (pdf_viewer.js:5162)
>     at continueRendering (pdf_viewer.js:5227)
>     at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:391)
>     at Object.onInvoke (core.js:17289)
>     at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:390)
>     at Zone.push../node_modules/zone.js/dist/zone.js.Zone.run (zone.js:150)
>     at resolvePromise (zone.js:831)
>     at zone.js:896
>     at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:423)
>     at Object.onInvokeTask (core.js:17280)
>     at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:422)
>     at Zone.push../node_modules/zone.js/dist/zone.js.Zone.runTask (zone.js:195)
>     at drainMicroTaskQueue (zone.js:601)
>     at ZoneTask.push../node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask [as invoke] (zone.js:502)
>     at invokeTask (zone.js:1744)
>     at WebSocket.globalZoneAwareCallback (zone.js:1770)
> ```

This is an well known issue:
https://github.com/VadimDez/ng2-pdf-viewer/issues/224

It seem that we need clean up the pdf-viewer before rerendering.

**Comment by boehlke** [comment]
@FinnStutzenstein 
Do you understand, why the checks fail? Seems to be something backend related... I didn't change anything in the backend.

**Comment by FinnStutzenstein** [comment]
Fixed travis. The server issues are gone now, I do not know what happend there. Please go on with this PR.

**Comment by emanuelschuetze** [comment]
I can't test the current PR. I run into this error if I try to open the projector list: 
```
Error: Uncaught (in promise): Error: Component PresentationControlComponent is not part of any NgModule or the module has not been imported into your module.
Error: Component PresentationControlComponent is not part of any NgModule or the module has not been imported into your module.
```

@boehlke @FinnStutzenstein Can you discuss please who can fix and continue with this PR. Thx.

**Comment by FinnStutzenstein** [comment]
@boehlke Hi, the controls are ready. Whats left todo is the slide (html, css) itself: Fullscreen of images, rotation of the PDF and the back background for images. Please see what you can do

**Comment by emanuelschuetze** [comment]
@boehlke Thanks again for your initial pull request! I'll merge it now. Feel free to contribute further improvements ;-).

**Pull request merged**